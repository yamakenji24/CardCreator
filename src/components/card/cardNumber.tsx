
export const CardNumber = (): JSX.Element => {
  return (
    <div className="absolute -right-16 -bottom-0">
      <div className="transform rotate-90 -translate-y-28 text-3xl font-bold text-center pb-11 w-72 text-white border-b-2 border-gray-400 border-dashed">
        â„– 014747
      </div>
    </div>
  )
}