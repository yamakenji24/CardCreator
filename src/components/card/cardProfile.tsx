
export const CardProfile = (): JSX.Element => {
  return (
    <div className="py-8 pr-8 pl-10">
      <div className="flex flex-row">
        <img
          src="https://github.com/yamakenji24.png"
          alt="yamakenji24"
          className="w-20 h-20 rounded-full"
        />
        <div className="m-0">
          <p className="text-3xl ml-10 mb-2 font-bold text-white">yamakenji24</p>
          <div className="mt-0 mr-0 mb-1 ml-4 text-gray-300 flex">
            <div className="h-4 w-4 mt-1 mr-2">
              <img src="/github.svg" alt="" />
            </div>
            yamakenji24
          </div>
        </div>
      </div>
    </div>
  );
};