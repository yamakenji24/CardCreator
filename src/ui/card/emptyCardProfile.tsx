
export const EmptyCardLayout = ({color}): JSX.Element => {
  return (
    <div className="w-crd h-80 my-40 mx-auto relative border-4 border-solid border-white p-1.5 rounded-3xl bg-gradient-to-r from-purple-400 via-pink-500 to-red-500">
      <div className={"w-full h-full rounded-3xl " + color}>
        <img src="/logo.png" className="w-24 h-24 rounded-full" />
      </div>
   </div>
  )
}