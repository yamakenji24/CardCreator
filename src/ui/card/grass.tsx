
interface Props {
  provider: string;
  name: string;
}

export const Grass = ({provider, name}: Props): JSX.Element => {
  if(provider !== 'github') return null;

  return (
    <div className="ml-28">
      <img
        src={"/_next/image?url=https%3A%2F%2Fgrass-graph.moshimo.works%2Fimages%2F"+name+".png%3Fbackground%3Dnone&w=1200&q=75"}
        width={380} height={70}
      />
    </div>
  )
}