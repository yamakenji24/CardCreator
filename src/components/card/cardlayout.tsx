import { CardNumber } from './cardNumber';
import { CardProfile } from './cardProfile';

export const CardLayout = (): JSX.Element => {
  return (
   <div className="w-crd h-80 my-40 mx-auto relative border-4 border-solid border-white p-1.5 rounded-3xl bg-gradient-to-r from-purple-400 via-pink-500 to-red-500">
      <div className="w-full h-full bg-black rounded-3xl">
        <CardProfile />
        <CardNumber />
      </div>
   </div>
  );
};