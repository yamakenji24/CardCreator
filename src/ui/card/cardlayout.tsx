import { Profile } from './profile/Profile';
import { useGetSession } from '../../utils/useGetSession';
import { OuterColorType } from '../../utils/useChangeColor';
import { EmptyCardLayout } from './emptyCardProfile';
import { Grass } from './grass';
import { isNullOrUndefined } from 'util';

interface Props {
  innerColor: string;
  outerColor: OuterColorType;
}

export const CardLayout = ({
  innerColor, outerColor
}: Props ): JSX.Element => {
  const {profileData} = useGetSession();
  if(isNullOrUndefined(profileData)) return <EmptyCardLayout innerColor={innerColor} outerColor={outerColor}/>
  return (
   <div id="cardScreen" className={"w-crd h-80 my-10 mx-auto relative border-4 border-solid border-white p-1.5 rounded-3xl bg-gradient-to-r from-"+outerColor.from+" via-"+outerColor.via+" to-"+outerColor.to}>
      <div className={"w-full h-full rounded-3xl bg-" + innerColor}>
        <div className="flex">
          <img src={"/_next/image?url=%2Flogo.svg&w=384&q=75"} className="rounded-full" width='100' height='100' />
          <Grass provider={profileData.provider} name={profileData.name} />
        </div>
        <Profile 
          name={profileData.name}
          picture={profileData.picture}
          provider={profileData.provider} 
          twitter={profileData.twitter}
          email={profileData.email}
          blog={profileData.blog}
        />
      </div>
   </div>
  );
};