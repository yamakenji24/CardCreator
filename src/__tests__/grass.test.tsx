import { render } from '@testing-library/react';
import { Grass } from '../ui/card/grass';

describe("<Grass />", () => {
  test("render grass img from existing projects", () => {
    const {asFragment} = render(<Grass provider="github" name="yamakenji24" />)
    expect(asFragment()).toMatchSnapshot()
  }),
  test("render null when provider is other than github", () => {
    const wrapper = render(<Grass provider="google" name="yamakenji24" />)
    expect(wrapper.container.innerHTML).toHaveLength(0);
  })
})