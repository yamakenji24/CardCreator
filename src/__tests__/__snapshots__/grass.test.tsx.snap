// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grass /> render grass img from existing projects 1`] = `
<DocumentFragment>
  <div
    class="ml-28"
  >
    <img
      height="70"
      src="/_next/image?url=https%3A%2F%2Fgrass-graph.moshimo.works%2Fimages%2Fyamakenji24.png%3Fbackground%3Dnone&w=1200&q=75"
      width="380"
    />
  </div>
</DocumentFragment>
`;
